src/api/assignment.ts

```diff
@@ -46,63 +44,62 @@ export class Assignment implements IAssignment {
         private _directoryPath: string,
         private _absoluteDirectoryPath: string,
         private _createdDate: Date,
-        private _adjustedAvailableDate: Date | null,
-        private _adjustedDueDate: Date | null,
+        private _availableDate: Date | null,
+        private _dueDate: Date | null,
         private _lastModifiedDate: Date,
+        private _stagedChanges: IStagedChange[],
 
         private _isDeferred: boolean,
         private _isExtended: boolean,
         private _isCreated: boolean,
         private _isAvailable: boolean,
         private _isClosed: boolean,
-        private _submissions?: ISubmission[],
-        private _stagedChanges?: IStagedChange[]
+
+        private _studentSubmissions: StudentSubmissions | undefined
     ) {}
+    get studentSubmissions() { return this._studentSubmissions }
     
     get id() { return this._id }
     get name() { return this._name }
     get directoryPath() { return this._directoryPath }
     get absoluteDirectoryPath() { return this._absoluteDirectoryPath }
     get createdDate() { return this._createdDate }
-    get adjustedAvailableDate() { return this._adjustedAvailableDate }
-    get adjustedDueDate() { return this._adjustedDueDate }
+    get availableDate() { return this._availableDate }
+    get dueDate() { return this._dueDate }
     get lastModifiedDate() { return this._lastModifiedDate }
+    get stagedChanges() { return this._stagedChanges }
     
 
     get isDeferred() { return this._isDeferred }
     get isExtended() { return this._isExtended }
     get isCreated() { return this._isCreated }
     get isAvailable() { return this._isAvailable }
     get isClosed() { return this._isClosed }
-
-    get submissions() { return this._submissions }
-    get activeSubmission() {
-        if (this._submissions === undefined) return undefined
-        if (this._submissions.length === 0) return null
-        return this._submissions.sort((a, b) => b.submissionTime.getTime() - a.submissionTime.getTime())[0]
-    }
-
-    get stagedChanges() { return this._stagedChanges }
     
 
     static fromResponse(data: AssignmentResponse): IAssignment {
+        let studentSubmissions: any = data.student_submissions ? {} : undefined
+        if (data.student_submissions) Object.keys(data.student_submissions).forEach((onyen) => {
+            studentSubmissions[onyen] = data.student_submissions![onyen].map((res) => Submission.fromResponse(res))
```

Comment:
How do we know `data.student_submissions` will always be present? Can we instead do a null check on it instead of forcing it to be dereferenced?